daisy/workflow/step_create_instances.go

```diff
@@ -100,6 +104,33 @@ func (c *CreateInstances) validate(w *Workflow) error {
 			if !diskValid(w, d) {
 				return fmt.Errorf("cannot create instance: disk not found: %s", d)
 			}
+			// Ensure disk is in the same project and zone.
+			if !isLink(d) {
```

Comment:
Do you need this check here? https://golang.org/pkg/regexp/#Regexp.FindStringSubmatch says that 'match' will be nil if 'd' doesn't match the URL regex.